main.tf

```diff
@@ -1,30 +1,24 @@
 locals {
-  access_token_json = var.use_ssm_parameter_store ? ({
+  access_token_secret_json = var.use_ssm_parameter_store ? ({
     "secrets" : [
       { "name" : "LaceworkAccessToken", "valueFrom" : local.ssm_parameter_arn }
     ]
-    }) : ({
-    "environment" : [
-      { "name" : "LaceworkAccessToken", "value" : var.lacework_access_token }
-    ]
-  })
-
-  server_url_json = length(var.lacework_server_url) > 0 ? (
-    var.use_ssm_parameter_store ? ({
-      "environment" : [
-        { "name" : "LaceworkServerUrl", "value" : var.lacework_server_url }
-      ]
-      }) : ({
-      "environment" : [
-        { "name" : "LaceworkAccessToken", "value" : var.lacework_access_token },
-        { "name" : "LaceworkServerUrl", "value" : var.lacework_server_url }
-      ]
-    })
-  ) : ({})
+  }) : ({})
+
+  environment_json = {
+    "environment" : [merge(
+      (! var.use_ssm_parameter_store) ? ({
+        "name" : "LaceworkAccessToken", "value" : var.lacework_access_token
+      }) : ({}),
+      length(var.lacework_server_url) > 0 ? ({
+        "name" : "LaceworkServerUrl", "value" : var.lacework_server_url
+      }) : ({}),
```

Comment:
This causes the same problem I ran into when trying to work on this.  The `merge` function will replace the values for the `name` and `value` keys if `use_ssm_parameter_store` is true and `lacework_server_url` length is greater than 0, rather than placing both objects in the list.

Here's the Terraform plan output of it replacing, rather than appending:
```sh
  # module.lacework_ecs.aws_ecs_task_definition.lacework_datacollector must be replaced
-/+ resource "aws_ecs_task_definition" "lacework_datacollector" {
      ~ arn                      = "arn:aws:ecs:us-east-1:713578388890:task-definition/lacework-ecs-task-7c416165:1" -> (known after apply)
      ~ container_definitions    = jsonencode(
          ~ [ # forces replacement
              ~ {
                  - cpu          = 0 -> null
                  ~ environment  = [
                      ~ {
                          ~ name  = "LaceworkAccessToken" -> "LaceworkServerUrl"
                          ~ value = "<access_token_redacted>" -> "https://api.lacework.net"
                        },
                    ]
```
From the Terraform docs (https://www.terraform.io/docs/language/functions/merge.html):
```
If more than one given map or object defines the same key or attribute, then the one that is later in the argument sequence takes precedence.
```

Reply:
What about join instead?

Reply:
This does seem to work:

![Screen Shot 2021-05-18 at 9 30 02 PM](https://user-images.githubusercontent.com/5712253/118756635-97357e00-b828-11eb-932b-62e99ec1c8ab.png)

Reply:
🔥 🔥 🔥 👍